File: QytetetJava/src/modeloqytetet/Calle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author jakerxd
 */
public class Calle extends Casilla{
	
	Calle(int numeroCasilla, TituloPropiedad titulo)
	{
		super(numeroCasilla, titulo.getPrecioCompra(), TipoCasilla.CALLE, titulo);
	}
	
	@Override
	protected TipoCasilla getTipo() {
		return TipoCasilla.CALLE;
	}
	
	@Override
	protected boolean soyEdificable() {
		return true;
	}
	
	@Override
	protected TituloPropiedad getTitulo() {
		return titulo;
	}
	
	public int pagarAlquiler()
	{
		return titulo.pagarAlquiler();
	}
	
	private void setTitulo(TituloPropiedad titulo){
		this.titulo = titulo;
	}
	
	public boolean tengoPropietario() {
		return titulo.tengoPropietario();
	}
	
	@Override
	public String toString() {
		return "Casilla{" + "numeroCasilla=" + super.getNumeroCasilla() + ", coste=" + super.getCoste() + ", tipo=" + getTipo() + ", titulo=" + getTitulo().toString() + '}';
	}
}

File: QytetetRuby/lib/calle.rb
# encoding: utf-8
module ModeloQytetet
	class Calle < Casilla
		def initialize(numero, titulo)
			super(numero, titulo.precioCompra, TipoCasilla::CALLE)
			@titulo = titulo
		end

		def asignarPropietario(jugador)
			@titulo.propietario = jugador

			return @titulo
		end

		def pagarAlquiler
			return @titulo.pagarAlquiler
		end

		def soyEdificable
			true
		end

		def tengoPropietario
			@titulo.tengoPropietario
		end

		def propietarioEncarcelado
			@titulo.propietarioEncarcelado
		end

		def to_s
			"Casilla{numeroCasilla=#{@numero}, coste=#{@coste}, tipo=#{@tipo}, titulo=#{@titulo.to_s()}"
		end
	end
end